### PR TITLE
Use netcoreapp and netstandard for targetgroups

### PR DESCRIPTION
**Pull request by chcosta**
Use netcoreapp and netstandard in place of netcoreapp1.1 and netstandard1.7 for src / ref projects.

/cc @weshaggard @ericstj @mellinoe

### CODE REVIEW

**Comment by chcosta** [comment]
I'll look at why this is failing.  Clearly, some instance was missed.

**Comment by chcosta** [comment]
I had intended to submit the reference project conversion separate from the source project change, but they too closely align for a handful of reference projects.  I've temporarily made a change to resolveContracts which allows the projects to build, but I'm assuming that this is likely to change after Wes' change is merged.  https://github.com/dotnet/corefx/pull/14525

File: src/Microsoft.Win32.Primitives/src/Microsoft.Win32.Primitives.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <IsPartialFacadeAssembly Condition="'$(TargetGroup)' == 'net463'">true</IsPartialFacadeAssembly>
     <!-- copy to uap10.1 until nuget makes the change to map that to ns1.7 -->
-    <PackageTargetFramework Condition="'$(TargetGroup)' == 'netstandard1.7'">netstandard1.7;uap10.1</PackageTargetFramework>
+    <PackageTargetFramework Condition="'$(TargetGroup)' == 'netstandard'">netstandard1.7;uap10.1</PackageTargetFramework>
```
Comment (weshaggard) [inline src/Microsoft.Win32.Primitives/src/Microsoft.Win32.Primitives.csproj]: @ericstj should this property become `<PackageTargetFramework Condition="'$(TargetGroup)' == 'netstandard'">$(PackageTargetFramework);uap10.1</PackageTargetFramework>`?

File: src/System.Collections/src/System.Collections.csproj [hunk 2]
```diff
@@ -8,23 +8,23 @@
     <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
     <!-- 436 is thrown because BitArray conflict with corelib -->
     <NoWarn>0436</NoWarn>
-    <PackageTargetFramework Condition="'$(TargetGroup)' == 'netstandard1.7'">netstandard1.7;uap10.1</PackageTargetFramework>
+    <PackageTargetFramework Condition="'$(TargetGroup)' == 'netstandard'">netstandard1.7;uap10.1</PackageTargetFramework>
```
Comment (weshaggard) [inline src/System.Collections/src/System.Collections.csproj]: `<PackageTargetFramework Condition="'$(TargetGroup)' == 'netstandard'">$(PackageTargetFramework);uap10.1</PackageTargetFramework>`

File: Tools-Override/resolveContract.targets [hunk 3]
```diff
@@ -30,7 +31,9 @@
     <ItemGroup Condition="'@(ContractProject)' != ''">
       <!--Don't flow the values for Configuration --> 
       <ContractProject> 
-        <UndefineProperties>Configuration;%(ContractProject.UndefineProperties)</UndefineProperties> 
+        <UndefineProperties Condition="'$(HasBuildConfigurations)' != 'true' and '%(ContractProject.TargetGroup)' == ''">Configuration;%(ContractProject.UndefineProperties)</UndefineProperties>
```
Comment (weshaggard) [inline Tools-Override/resolveContract.targets]: I pushed an update to https://github.com/dotnet/corefx/pull/14525 feel free to merge when green (I will if I'm online later).

Comment (weshaggard) [inline Tools-Override/resolveContract.targets]: I've merged so when you fix the conflicts you should be able to undo any changes to this file.

**Comment by mellinoe** [comment]
Are we going to change the default `BuildConfiguration` to have `netcoreapp` instead of `netcoreapp1.1` ?

**Comment by chcosta** [comment]
@mellinoe , I've changed the default in this PR.  

I'll take a look at merging Wes' changes now

**Comment by mellinoe** [comment]
> @mellinoe , I've changed the default in this PR.

Oops, dunno how I missed that.

**Pull request merged**